dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/ZonedIntervalIterable.java

```diff
@@ -94,31 +90,20 @@ public static ZonedIntervalIterable forConnectorArguments(@Nonnull ConnectorArgu
    * end ).
    *
    * @param arguments connector arguments
-   * @param timeUnit the length of the intervals. Only supports {@value #SUPPORTED_UNITS} at the
-   *     moment
+   * @param duration the length of the intervals
    * @return a nonnull ZonedIntervalIterable
    * @throws MetadataDumperUsageException in case of arguments incompatibility or missing arguments
-   * @throws IllegalArgumentException if any {@link ChronoUnit} other than `HOURS` or `DAYS` is
-   *     passed
+   * @throws IllegalArgumentException if then {@link Duration} is more than a day or does not divide
+   *     a day evenly
    */
   @Nonnull
   public static ZonedIntervalIterable forConnectorArguments(
-      @Nonnull ConnectorArguments arguments, @Nonnull ChronoUnit timeUnit)
+      @Nonnull ConnectorArguments arguments, @Nonnull Duration duration)
       throws MetadataDumperUsageException {
-    int unitsInADay;
-
-    switch (timeUnit) {
-      case DAYS:
-        unitsInADay = 1;
-        break;
-      case HOURS:
-        unitsInADay = 24;
-        break;
-      default:
-        throw new IllegalArgumentException(
-            String.format(
-                "`timeUnit=%s` is not supported. Only allowed units are: %s",
-                timeUnit, SUPPORTED_UNITS));
+    if (!isValidDuration(duration)) {
```

Comment:
You could simplify this with [`Preconditions.checkArgument()`](https://guava.dev/releases/19.0/api/docs/com/google/common/base/Preconditions.html#checkArgument(boolean,%20java.lang.String,%20java.lang.Object...)).